onebusaway-android/src/main/java/org/onebusaway/android/ui/TripPlanActivity.java

```diff
@@ -0,0 +1,429 @@
+
+        // see if there is data from intent
+        Intent intent = getIntent();
+        if (intent != null && intent.getExtras() != null
+                && intent.getBooleanExtra(INTENT_SOURCE, false)) {
+
+            ArrayList<Itinerary> itineraries = (ArrayList<Itinerary>) intent.getExtras().getSerializable(OTPConstants.ITINERARIES);
+
+            if (itineraries != null) {
+                bundle.putSerializable(OTPConstants.ITINERARIES, itineraries);
+                newItineraries = true;
+            }
+
+            if (intent.getIntExtra(PLAN_ERROR_CODE, -1) != -1) {
+                bundle.putSerializable(SHOW_ERROR_DIALOG, true);
+                bundle.putInt(PLAN_ERROR_CODE, intent.getIntExtra(PLAN_ERROR_CODE, 0));
+                bundle.putString(PLAN_ERROR_URL, intent.getStringExtra(PLAN_ERROR_URL));
+            }
+
+            setIntent(null);
+            bundle.putBoolean(REQUEST_LOADING, false);
+            mRequestLoading = false;
+
+        }
+
+        if (mRequestLoading || bundle.getBoolean(REQUEST_LOADING)) {
+            showProgressDialog();
+        }
+
+        // Check which fragment to create
+        boolean haveTripPlan = bundle.getSerializable(OTPConstants.ITINERARIES) != null;
+
+        TripPlanFragment fragment = (TripPlanFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.trip_plan_fragment_container);
+        if (fragment == null) {
+            fragment = new TripPlanFragment();
+            fragment.setArguments(bundle);
+            getSupportFragmentManager().beginTransaction()
+                    .add(R.id.trip_plan_fragment_container, fragment).commit();
+        }
+
+        mPanel = (SlidingUpPanelLayout) findViewById(R.id.trip_plan_sliding_layout);
+
+        mPanel.setEnabled(haveTripPlan);
+
+        if (haveTripPlan) {
+            initResultsFragment();
+            if (bundle.getBoolean(PANEL_STATE_EXPANDED) || newItineraries) {
+                mPanel.setPanelState(SlidingUpPanelLayout.PanelState.EXPANDED);
+            }
+        }
+
+        // show error dialog if necessary
+        if (bundle.getBoolean(SHOW_ERROR_DIALOG)) {
+            int planErrorCode = intent.getIntExtra(PLAN_ERROR_CODE, 0);
+            String planErrorUrl = intent.getStringExtra(PLAN_ERROR_URL);
+            showFeedbackDialog(planErrorCode, planErrorUrl);
+        }
+
+        // Set the height of the panel after drawing occurs.
+        final ViewGroup layout = (ViewGroup)findViewById(R.id.trip_plan_fragment_container);
+        ViewTreeObserver vto = layout.getViewTreeObserver();
+        vto.addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+            @Override
+            public void onGlobalLayout() {
+                layout.getViewTreeObserver().removeGlobalOnLayoutListener(this);
+                int viewHeight = findViewById(android.R.id.content).getHeight();
+                int height = layout.getMeasuredHeight();
```

Comment:
I thought that changing this to `layout.getHeight()` would resolve the problem (see http://stackoverflow.com/a/19524365/937715), but it doesn't seem to have any effect for me.  I'll keep looking...